🤽📖 add Bounty process definition (#226)
leapdao/meta: playbook/bounties.md

  * [Funding Circle](#funding-circle)
  * [Bounty owner/gardener](#bounty-ownergardener)
  * [Gain for the Role](#gain-for-the-role)
  * [Size](#size)
* [Who can create a new bounty](#who-can-create-a-new-bounty)
* [How to create a new bounty](#how-to-create-a-new-bounty)
* [Pair bounty](#pair-bounty)
* [Objecting bounties](#objecting-bounties)
* [Work on Bounty](#work-on-bounty)
  * [Gardener](#gardener)
  * [Worker](#worker)
  * [Reviewer](#reviewer)
* [Challenging bounties](#challenging-bounties)
* [Links](#links)

## Glossary

[Role](https://www.holacracy.org/constitution#art11), [Action](https://www.holacracy.org/constitution#art122), [Project](https://www.holacracy.org/constitution#art122) and [Circle](https://www.holacracy.org/constitution#art21) are all concepts defined in Holacracy Constitution.

## Bounty definition

Bounty is a reward for completing a properly defined Action or Project within a Circle. For simplicity, we will may call such Actions or Projects themselves a Bounties from now on.

Properly defined bounty must specify the following attributes

### Title

Succinct name for the bounty.

Example:
> Implement Basic Token Governance UI

### Scope

> A list of specific things which should be done to deliver the bounty. These could be seen as requirements to verify/review bounty against.

Reviewer: ```suggestion
A list of specific things which should be done to deliver the bounty. These could be seen as requirements to verify/review the bounty against.
```

### Deliverables

Artifacts produced as the result of this bounty. Something that could be verified/reviewed. Some examples: updated code, deployment made, blog post published, public event conducted etc

### Funding Circle

The Circle to fund this bounty

### Bounty owner/gardener

The Role who is responsible for the bounty. The Role should belong to Funding Circle.

### Gain for the Role

> How the completion of this bounty helps to pursue the Role's purpose

Reviewer: ```suggestion
How the completion of this bounty helps to pursue the Role's purpose.
```

### Size

Size of the bounty.

We use t-shirt sizes:

* XS. 200 DAI
* S. 350 DAI
* M. 550 DAI
* L. 900 DAI

> * XL. 1400 DAI

Reviewer: ```suggestion
* size-XS: 200 DAI
* size-S: 350 DAI
* size-M: 550 DAI
* size-L: 900 DAI
* size-XL: 1400 DAI
```

## Who can create a new bounty

Any Role within the Organization. People without Roles may also propose a bounty on behalf of some Role.

## How to create a new bounty

1. Create a Github issue with properly defined bounty description (see above) in an appropriate repository.
2. If the bounty spans across multiple repositories, consider splitting it in a smaller per-repo bounties if possible or put in `leapdao/meta` repository.
3. Submit proposal via the [bounty form](http://bounty.leapdao.org/viewform)
4. Add the bounty to the [bounties board](https://github.com/orgs/leapdao/projects/6)

## Pair bounty

If two people work on the bounty together, the payout increases by 1.5x.

## Objecting bounties

Any bounty can be objected by anyone through Slack or GitHub for 2 days after its creation. A valid objection should describe how the bounty is not helping bounty Role to achieve it's purpose or how the bounty is harming the Organization. The objector must ensure that their objection is clearly communicated to the bounty proposer and e.g. not misunderstood as a minor comment on the issue.

> In the cases where objector and proposer are not able to come to an acceptable solution, the Circle’s facilitator must be involved to enact Integrative Decision-Making Process as specified in Holacracy constitution ([§1.3.5](https://www.holacracy.org/constitution#art135)) to solve the objection as a tension.

Reviewer: ```suggestion
In the cases where objector and proposer are not able to come to an acceptable solution, the Circle’s facilitator must be involved to enact Integrative Decision-Making Process as specified in Holacracy constitution ([§3.3.5](https://www.holacracy.org/constitution#art335)) to solve the objection as a tension.
```

Author: great catch!

If the tension was integrated successfully or if there weren’t any public objections, the bounty is considered approved.

## Expiration

All non-taken bounties older than 1 month (time of submission through bounty form), MUST be submitted to bounty form again and undergo a new round of objections (as defined in by this spec) before the work can be started on it.

## Work on Bounty

### Gardener

Gardener is the one who creates the bounty and is responsible for it's completion.

Gardener is expected to:

* find a Worker for the Bounty
* help the Worker to understand the scope of the Bounty
* find a Reviewer for the Bounty.
* ensure the Bounty doesn’t get stalled (see "Expiration" section)

> * request the payout for the Bounty once it is done and reviewed (see "Payout" section below)

Reviewer: ```suggestion
* find a Worker for the Bounty.
* help the Worker to understand the scope of the Bounty.
* find a Reviewer for the Bounty.
* ensure the Bounty doesn’t get stalled (see "Expiration" section).
* request the payout for the Bounty once it is done and reviewed (see "Payout" section below).
```

If worker or reviewer are not publishing any updates for the bounty for 2 working days, then anyone can challenge him and take over the worker role. Updates for worker role can be in any form, most notably: commits to WIP Pull Request, updates on product artifact (hackmd, presentation, diagram etc), review comments, updates in form of github comments or slack conversations.

### Challenging gardener
If gardener is not helping worker to find a reviewer or doesn't address worker's questions within 2 days, then anyone can challenge him and take over a gardener role on the bounty.

## Payout

Funds allocated for bounty are split between Gardener, Worker and Reviewer in a proportion defined by Gardener (can be negotiated).

It is Gardener's duty to request the payout once the bounty is completed and reviewed. Payout is requested on Slack in a Circle's channel like this:
  > 🌴💰 Requesting payout for **\<bounty title\>** \
  > Issue: \<link to the Bounty on Github\> \
  > Product: \<link to deliverable, e.g. PR or writeup\> \
  > gardener: \<@gardener\> \<gardener share\> \
  > worker: \<@worker\> \<worker share\> \
  > reviewer: \<@reviewer\> \<reviewer share\>

## Links

> * [Bounty issue template](https://github.com/leapdao/leap-node/blob/master/.github/ISSUE_TEMPLATE/bounty.md)

Reviewer: ```suggestion
* [Bounty issue template](https://github.com/leapdao/leap-node/blob/master/.github/ISSUE_TEMPLATE.md)
```

Author: that was on purpose :) Github allows many different issue templates — in this case we are creating the one for "Bounty"

Example:
![image](https://user-images.githubusercontent.com/163447/77681625-ae6f3780-6fa6-11ea-9cd3-f097e1e2951c.png)

Author: again.. the URL is such on purpose :)

Reviewer: ok, does it link to `leap-node` repo on purpose? because there is no such file in that repo, and no open PR to add it.

> * [Bounty issue template](https://github.com/leapdao/leap-node/blob/master/.github/ISSUE_TEMPLATE/bounty.md)

Author: Certainly not. It should link to the file created in this PR

```suggestion
* [Bounty issue template](https://github.com/leapdao/meta/blob/master/.github/ISSUE_TEMPLATE/bounty.md)
```
